actors/miner/tests/miner_actor_test_precommit_batch.rs

```diff
@@ -0,0 +1,320 @@
+        );
+    }
+
+    #[test]
+    fn one_bad_apple_ruins_batch() {
+        // This test does not enumerate all the individual conditions that could cause a single precommit
+        // to be rejected. Those are covered in the PreCommitSector tests, and we know that that
+        // method is implemented in terms of a batch of one.
+
+        let period_offset = ChainEpoch::from(100);
+
+        let h = ActorHarness::new(period_offset);
+        let mut rt = h.new_runtime();
+        rt.set_balance(TokenAmount::from(BIG_BALANCE));
+        rt.set_received(TokenAmount::zero());
+
+        let precommit_epoch = period_offset + 1;
+        rt.set_epoch(precommit_epoch);
+        h.construct_and_verify(&mut rt);
+        let dl_info = h.deadline(&rt);
+
+        let sector_expiration =
+            dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period;
+        let sectors = vec![
+            h.make_pre_commit_params(100, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(101, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(102, precommit_epoch - 1, rt.epoch, vec![]), // Expires too soon
+        ];
+
+        expect_abort_contains_message(
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "sector expiration",
+            h.pre_commit_sector_batch(
+                &mut rt,
+                PreCommitSectorBatchParams { sectors },
+                &PreCommitBatchConfig { sector_weights: vec![], first_for_miner: true },
+                BigInt::zero(),
+            ),
+        );
+    }
+
+    #[test]
+    fn duplicate_sector_rejects_batch() {
+        let period_offset = ChainEpoch::from(100);
+
+        let h = ActorHarness::new(period_offset);
+        let mut rt = h.new_runtime();
+        rt.set_balance(TokenAmount::from(BIG_BALANCE));
+        rt.set_received(TokenAmount::zero());
+
+        let precommit_epoch = period_offset + 1;
+        rt.set_epoch(precommit_epoch);
+        h.construct_and_verify(&mut rt);
+        let dl_info = h.deadline(&rt);
+
+        let sector_expiration =
+            dl_info.period_end() + DEFAULT_SECTOR_EXPIRATION * rt.policy.wpost_proving_period;
+        let sectors = vec![
+            h.make_pre_commit_params(100, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(101, precommit_epoch - 1, sector_expiration, vec![]),
+            h.make_pre_commit_params(100, precommit_epoch - 1, sector_expiration, vec![]), // Duplicated
+        ];
+
+        expect_abort_contains_message(
+            ExitCode::USR_ILLEGAL_ARGUMENT,
+            "duplicate sector number 100",
+            h.pre_commit_sector_batch(
+                &mut rt,
+                PreCommitSectorBatchParams { sectors },
+                &PreCommitBatchConfig { sector_weights: vec![], first_for_miner: true },
+                BigInt::zero(),
+            ),
+        );
```

Comment:
nit: decorative rt.reset() to signal that the mock runtime expectations are out of wack here